tsl/src/compression/compress_utils.c

```diff
@@ -420,7 +430,7 @@ tsl_compress_chunk_wrapper(Chunk *chunk, bool if_not_compressed)
 		ereport((if_not_compressed ? NOTICE : ERROR),
 				(errcode(ERRCODE_DUPLICATE_OBJECT),
 				 errmsg("chunk \"%s\" is already compressed", get_rel_name(chunk->table_id))));
-		return false;
+		return if_not_compressed ? true : false;
```

Comment:
if `if_not_compressed` is false we do ereport(ERROR) which will not return
so the return value here will always be true, also the ternary operator is redundant and makes you pause why it is here